File: programming/lab2/src/main/java/moves/special/DazzlingGleam.java
package moves.special;

import ru.ifmo.se.pokemon.Pokemon;
import ru.ifmo.se.pokemon.SpecialMove;
import ru.ifmo.se.pokemon.Type;

/**
 * Inflicts regular damage with no additional effect.
 */
public class DazzlingGleam extends SpecialMove {
    public DazzlingGleam() {
        super(Type.FAIRY, 80, 100);
    }

    @Override
    protected boolean checkAccuracy(Pokemon pokemon, Pokemon pokemon1) {
        return true;
    }

    @Override
    protected String describe() {
        return "использует атаку Dazzling Gleam";
    }
}

File: programming/lab2/app/src/main/kotlin/org/itmo/moves/special/DazzlingGleam.kt
package org.itmo.moves.special

import ru.ifmo.se.pokemon.Pokemon
import ru.ifmo.se.pokemon.SpecialMove
import ru.ifmo.se.pokemon.Type

class DazzlingGleam : SpecialMove(Type.FAIRY, 80.0, 100.0) {
    override fun checkAccuracy(p0: Pokemon?, p1: Pokemon?): Boolean {
        return true
    }

    override fun describe(): String {
        return "использует Dazzling Gleam"
    }
}